numpy/core/setup.py

```diff
@@ -965,6 +965,19 @@ def generate_umath_c(ext, build_dir):
                                                  generate_umath.__file__))
         return []
 
+    def generate_umath_doc_header(ext, build_dir):
+        target = join(build_dir, header_dir, '_umath_doc_generated.h')
+        dir = os.path.dirname(target)
+        if not os.path.exists(dir):
+            os.makedirs(dir)
+
+        generate_umath_doc_py = join(codegen_dir, 'generate_umath_doc.py')
+        if newer(generate_umath_doc_py, target):
+            n = dot_join(config.name, 'generate_umath_doc')
+            generate_umath_doc = npy_load_module(
```

Comment:
This file seems to be the only place that uses `npy_load_module`, and that function uses tools that will be deprecated in Python 3.12. I wonder if we should use the alternative `numpy.distutils.misc_util.exec_mod_from_location` instead?

Reply:
I have replaced the both cases of `npy_load_module` to `exec_mod_from_location`.